src/benchmark/tpcds_benchmark.cpp

```diff
@@ -0,0 +1,142 @@
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
+
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <string>
+
+#include "SQLParser.h"
+#include "SQLParserResult.h"
+#include "benchmark_runner.hpp"
+#include "cli_config_parser.hpp"
+#include "cxxopts.hpp"
+#include "file_based_benchmark_item_runner.hpp"
+#include "file_based_table_generator.hpp"
+#include "json.hpp"
+#include "scheduler/current_scheduler.hpp"
+#include "scheduler/node_queue_scheduler.hpp"
+#include "scheduler/topology.hpp"
+#include "sql/sql_pipeline.hpp"
+#include "sql/sql_pipeline_builder.hpp"
+#include "sqlite_add_indices.hpp"
+#include "storage/chunk_encoder.hpp"
+#include "utils/assert.hpp"
+#include "visualization/lqp_visualizer.hpp"
+#include "visualization/pqp_visualizer.hpp"
+
+using namespace opossum;  // NOLINT
+
+bool data_files_available(std::string table_path);
+
+int main(int argc, char* argv[]) {
+  auto cli_options = opossum::BenchmarkRunner::get_basic_cli_options("TPC-DS Benchmark");
+
+  // clang-format off
+  cli_options.add_options()
+    ("s,scale", "Database scale factor (1 ~ 1GB)", cxxopts::value<int>()->default_value("1"));
+  // clang-format on
+
+  std::shared_ptr<opossum::BenchmarkConfig> config;
+  int scale_factor;
+
+  if (opossum::CLIConfigParser::cli_has_json_config(argc, argv)) {
+    // JSON config file was passed in
+    const auto json_config = opossum::CLIConfigParser::parse_json_config_file(argv[1]);
+    scale_factor = json_config.value("scale", 1);
+    config = std::make_shared<opossum::BenchmarkConfig>(
+        opossum::CLIConfigParser::parse_basic_options_json_config(json_config));
+
+  } else {
+    // Parse regular command line args
+    const auto cli_parse_result = cli_options.parse(argc, argv);
+
+    if (CLIConfigParser::print_help_if_requested(cli_options, cli_parse_result)) {
+      return 0;
+    }
+    scale_factor = cli_parse_result["scale"].as<int>();
```

Comment:
What happens if I pass in `0.9`? Does this get erroneously rounded to `1` and accepted?

Reply:
This happens:
```
terminate called after throwing an instance of 'cxxopts::argument_incorrect_type'
  what():  Argument ‘0.9’ failed to parse
```